packages/block-editor/src/components/inspector-controls-tabs/index.js

```diff
@@ -115,11 +115,15 @@ export default function InspectorControlsTabs( {
 		}
 	}
 
+	const blockType = getBlockType( blockName );
+
 	// We have multiple tabs with contents so render complete TabPanel.
 	return (
 		<TabPanel
 			className="block-editor-block-inspector__tabs"
 			tabs={ availableTabs }
+			initialTabName={ blockType.defaultTab }
+			key={ clientId }
```

Comment:
Open to better approaches here to force the tab panel to reflect different block types' default tabs.